build-support/githooks/pre-commit

```diff
@@ -53,7 +53,7 @@ echo "* Checking shell scripts via our custom linter"
 # fails in pants changed.
 if git rev-parse --verify "${MERGE_BASE}" &>/dev/null; then
   echo "* Checking imports"
-  ./build-support/bin/isort.sh || die "To fix import sort order, run \`build-support/bin/isort.sh -f\`"
+  ./pants lint-v2 :: || die "To fix import sort order, run \`./pants fmt-v2 ::\`"
```

Comment:
Given that we're unconditionally running this here, does this mean the line above does not need to filter `nolint` targets? Or should this be filtering `nolint` as well?

Reply:
Additionally, although caching will make this reasonably fast, running something like `--changed-parent="$(git_merge_base)"` would still be faster.

Having said that: if you're going to lint everything like this, you should remove it from the `if git rev-parse --verify "${MERGE_BASE}" &>/dev/null; then` block here, since it will no longer require the branch to diff against.

Reply:
Went back to using `--changed-parent="$(git_merge_base)"`. Thanks for explaining the reason we had that!